src/balancerd/src/lib.rs

```diff
@@ -595,11 +611,7 @@ impl PgwireBalancer {
 
         // Now blindly shuffle bytes back and forth until closed.
         // TODO: Limit total memory use.
-        // Ignore error returns because they are not actionable, and not even useful to record
-        // metrics of. For example, running psql in a shell then exiting with ctrl+D produces an
-        // error, even though it was an intended exit by the user. Those connections should not get
-        // recorded as errors, as that's probably a misleading metric.
-        let _ = tokio::io::copy_bidirectional(&mut client_counter, &mut mz_stream).await;
+        let conn_res = tokio::io::copy_bidirectional(&mut client_counter, &mut mz_stream).await;
```

Comment:
Part of me wants to return an error when this is an error and then handle logging in `handle_connection` like we do for the adapter.

Reply:
That makes a ton of sense to me!